mail/permissions.py

```diff
@@ -38,15 +39,43 @@ def has_object_permission(self, request, view, obj):
         Returns:
             boolean
         """
-        sender_user = request.user
-        if not has_permission(sender_user, Permissions.CAN_MESSAGE_LEARNERS):
+        if not obj.email_optin:
```

Comment:
The options for email optin on the form are `"Faculty, staff, and other learners can send me emails"` and `"I don't want to receive any emails"`